pulsar-client/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java

```diff
@@ -154,6 +154,20 @@
      */
     ConsumerBuilder<T> ackTimeout(long ackTimeout, TimeUnit timeUnit);
 
+    /**
+     * Set the timeout for unacked messages, truncated to the nearest millisecond. The timeout needs to be greater than
+     * 10 seconds.
+     *
+     * @param ackTimeout
+     *            for unacked messages.
+     * @param tickDuration
+     *            granularity of timeout checking for unacked messages. tickDuration influence timeout accuracy, if set
+     *            ackTimeout = 10 and tickDuration = 2, actual ackTimeout will be 10 to 12.
+     * @param timeUnit
+     *            unit in which the timeout is provided.
+     */
+    ConsumerBuilder<T> ackTimeout(long ackTimeout, long tickDuration, TimeUnit timeUnit);
```

Comment:
I'd rather not expose this in API, unless there are strong reason to chose different values of `tickDuration`. I'd prefer that we just pick a reasonable value in the implementation that can work well in most cases, just to avoid overloading the user with too many options.

Reply:
I'm already set the tickDuration to 1000 mills